ocaml/typing/typecore.ml

```diff
@@ -6982,7 +6985,8 @@ and type_cases
                    match compiler, thread it through into typedtree. *)
                 let { arg; sort; cases; partial = _ } =
                   type_match
```

Comment:
Let's leave a short comment that explains why this is ok.

Reply:
Done!

Reply:
Updated to do both of these things, together with adding a comment in the part of the function where `require_value_cases` is used